src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -263,18 +272,60 @@ public void renameFileToName(String targetFileName) {
         }
     }
 
-    private void performRenameWithConflictCheck(String targetFileName) {
+    public void performRenameWithConflictCheck(String targetFileName) {
         Optional<Path> existingFile = linkedFileHandler.findExistingFile(linkedFile, entry, targetFileName);
         boolean overwriteFile = false;
 
         if (existingFile.isPresent()) {
-            overwriteFile = dialogService.showConfirmationDialogAndWait(
-                    Localization.lang("File exists"),
-                    Localization.lang("'%0' exists. Overwrite file?", targetFileName),
-                    Localization.lang("Overwrite"));
+            Path existingFileVal = existingFile.get();
+            // show when hovering over "Keep both" button
+            String suggestedFileName = FileNameUniqueness.getNonOverWritingFileName(
+                existingFileVal.getParent(),
+                existingFileVal.getFileName().toString()
+            );
+
+            Label contentLabel = new Label(Localization.lang("Target file name: \n'%0'", targetFileName));
+            contentLabel.setWrapText(true);
+            DialogPane dialogPane = new DialogPane();
+            dialogPane.setPrefSize(700, 200);
+            dialogPane.setContent(contentLabel);
+
+            ButtonType overrideButton = new ButtonType("Override", ButtonBar.ButtonData.OTHER);
```

Comment:
Add proper Localization for these button types here as well using Localization.lang